File: decompiled/3.13.1/be/ehealth/businessconnector/chapterIV/exception/ChapterIVBusinessConnectorException.java
package be.ehealth.businessconnector.chapterIV.exception;

import be.ehealth.technicalconnector.exception.ConnectorException;
import be.fgov.ehealth.errors.soa.v1.SOAErrorType;
import java.text.MessageFormat;

public class ChapterIVBusinessConnectorException extends ConnectorException {
   private static final long serialVersionUID = 8040708257962683258L;
   private SOAErrorType cause;

   public ChapterIVBusinessConnectorException(ChapterIVBusinessConnectorExceptionValues errorCodeValue, SOAErrorType causeError, Object... params) {
      super(MessageFormat.format(errorCodeValue.getMessage(), params), errorCodeValue.getErrorCode());
      this.cause = causeError;
   }

   public ChapterIVBusinessConnectorException(ChapterIVBusinessConnectorExceptionValues errorCodeValue, Throwable e, SOAErrorType causeError, Object... params) {
      super(MessageFormat.format(errorCodeValue.getMessage(), params), errorCodeValue.getErrorCode(), e);
      this.cause = causeError;
   }

   public ChapterIVBusinessConnectorException(ChapterIVBusinessConnectorExceptionValues errorCodeValue, SOAErrorType causeError) {
      super(errorCodeValue.getMessage(), errorCodeValue.getErrorCode());
      this.cause = causeError;
   }

   public ChapterIVBusinessConnectorException(ChapterIVBusinessConnectorExceptionValues errorCodeValue, Throwable e, Object... params) {
      super(MessageFormat.format(errorCodeValue.getMessage(), params), errorCodeValue.getErrorCode(), e);
   }

   public ChapterIVBusinessConnectorException(ChapterIVBusinessConnectorExceptionValues error, String param) {
      this(error, (SOAErrorType)null, param);
   }

   public SOAErrorType getSOAError() {
      return this.cause;
   }
}

File: src/main/kotlin/org/taktik/connector/business/chapterIV/exception/ChapterIVBusinessConnectorException.kt
package org.taktik.connector.business.chapterIV.exception

import be.fgov.ehealth.errors.soa.v1.SOAErrorType
import org.taktik.connector.technical.exception.ConnectorException
import java.text.MessageFormat

class ChapterIVBusinessConnectorException : ConnectorException {
    var soaError: SOAErrorType? = null
        private set

    constructor(errorCodeValue: ChapterIVBusinessConnectorExceptionValues,
                causeError: SOAErrorType?,
                vararg params: Any) : super(MessageFormat.format(errorCodeValue.message, *params), errorCodeValue.errorCode) {
        this.soaError = causeError
    }

    constructor(errorCodeValue: ChapterIVBusinessConnectorExceptionValues,
                e: Throwable,
                causeError: SOAErrorType?,
                vararg params: Any) : super(MessageFormat.format(errorCodeValue.message, *params), errorCodeValue.errorCode, e) {
        this.soaError = causeError
    }

    constructor(errorCodeValue: ChapterIVBusinessConnectorExceptionValues,
                causeError: SOAErrorType?) : super(errorCodeValue.message, errorCodeValue.errorCode) {
        this.soaError = causeError
    }

    constructor(errorCodeValue: ChapterIVBusinessConnectorExceptionValues,
                e: Throwable,
                vararg params: Any) : super(MessageFormat.format(errorCodeValue.message, *params), errorCodeValue.errorCode, e)

    constructor(error: ChapterIVBusinessConnectorExceptionValues,
                param: String) : this(error, null as SOAErrorType?, param)

}
